### PR TITLE
JP-2988: Update ramp_fit tests for change in saturation flag propagation

### PR DESCRIPTION
**Pull request by kmacdonald-stsci**
Resolves [JP-2988](https://jira.stsci.edu/browse/JP-2988)

Closes #

This PR addresses the updated way the saturation flags propagate through to the rateints and rates product.  This update affected CI tests that needed to be changed to reflect the new way the saturation flag gets set.  In the rateints product, the saturation flag only gets set if all groups in an integration are saturated; in this case the integration also gets flagged with the DO_NOT_USE flag.  In the rates product, the saturation flag only gets set if all integrations are marked SATURATED; in this instance, the pixel also gets flagged as DO_NOT_USE.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Review by hbushouse — APPROVED** [review]
Code changes look "GOOD" to me. Add a change log entry (in the jwst repo) that references the actual ramp_fit code changes in the stcal PR, so that we have a record of that on the jwst side.

**Comment by hbushouse** [comment]
I assume all of the CI failures won't go away until the changes in stcal get merged and released.

**Comment by kmacdonald-stsci** [comment]
> I assume all of the CI failures won't go away until the changes in stcal get merged and released.

Yes.  The STCAL PR https://github.com/spacetelescope/stcal/pull/125 will need to be merged first.  Further, even after merging, I expect many regression test failures due to the new data quality propagation rules.

File: CHANGES.rst [hunk 1]
```diff
@@ -85,6 +85,12 @@ photom
   of NIRSpec IFU data with EXTENDED source type; correct units in pixel area
   division to sr from square arcseconds [#7336]
 
+ramp_fitting
+--------
```
Comment (hbushouse) [inline CHANGES.rst:89]: Still need to fix the underline here, so that it matches the length of the text above.

File: CHANGES.rst [hunk 2]
```diff
@@ -85,6 +85,12 @@ photom
...
+
+- Change the propagation of the SATURATED flag to be done only for complete
+  saturation. [#7363]
```
Comment (hbushouse) [inline CHANGES.rst:92]: Oh, and also reference the stcal PR number here, in addition to 7363.

**Comment by hbushouse** [comment]
stcal 1.2.2 has just been released, which contains the updates to the ramp_fit step needed to support this PR. So add to this PR a change to the setup.cfg file that pins stcal to >=1.2.2, and then this should be mergeable.

File: CHANGES.rst (hunk 1 above)
Comment (kmacdonald-stsci) [reply CHANGES.rst]: Fixed.

File: CHANGES.rst (hunk 2 above)
Comment (kmacdonald-stsci) [reply CHANGES.rst]: I linked the STCAL PR number here.

**Comment by kmacdonald-stsci** [comment]
> stcal 1.2.2 has just been released, which contains the updates to the ramp_fit step needed to support this PR. So add to this PR a change to the setup.cfg file that pins stcal to >=1.2.2, and then this should be mergeable.

I updated the setup.cfg file to reflect the new STCAL release.

**Comment by hbushouse** [comment]
I believe most of the failing CI tests are due to recent changes in the way the "cv2" package is imported, which may not be included in this branch yet.

**Comment by hbushouse** [comment]
Checking with @zacharyburnett to see if the change to the way `cv2` is imported in the jump step code in `stcal` might be causing these problems.

**Comment by zacharyburnett** [comment]
It looks like this slipped past the downstream tests for `stcal`; the issue is that `stcal` is not unconditionally requiring `opencv-python`, but the `jump_step` tests are using `stcal` code that makes ellipses / circles with it. I've made a PR #7377 that adds it to testing, and also conditionally skips these tests if it's not installed.

**Comment by hbushouse** [comment]
@zacharyburnett So once #7377 is merged and we then rebase this PR, the errors should go away (theoretically)?

**Comment by zacharyburnett** [comment]
@hbushouse yes, assuming the tests pass over there

**Comment by hbushouse** [comment]
Yay, CI tests look good now, except for the unrelated errors in xdist. So I'm merging.

**Comment by jdavies-st** [comment]
This PR is connect to the merging of spacetelescope/stcal#125

**Pull request merged**